[libmodplug] Build error on wasm32-emscripten

Package: libmodplug:wasm32-emscripten@0.8.9.0#10

**Host Environment**

- Host: arm64-osx
- Compiler: Clang 19.0.0
-    vcpkg-tool version: 2024-01-11-710a3116bbd615864eef5f9010af178034cb9b44
    vcpkg-scripts version: 215a25355 2024-03-01 (3 hours ago)

**To Reproduce**

`vcpkg install libmodplug`

**Failure logs**

```
-- Downloading https://github.com/Konstanty/libmodplug/archive/5a39f5913d07ba3e61d8d5afdba00b70165da81d.tar.gz -> Konstanty-libmodplug-5a39f5913d07ba3e61d8d5afdba00b70165da81d.tar.gz...
-- Extracting source <path to vcpkg>/downloads/Konstanty-libmodplug-5a39f5913d07ba3e61d8d5afdba00b70165da81d.tar.gz
-- Applying patch 001-automagically-define-modplug-static.patch
-- Applying patch 002-detect_sinf.patch
-- Applying patch 003-use-static-cast-for-ctype.patch
-- Applying patch 004-export-pkgconfig.patch
-- Applying patch 005-fix-install-paths.patch
-- Using source at <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean
-- Found external ninja('1.11.1').
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /Applications/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j11
    Working Directory: <path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg
    See logs for more information:
      <path to vcpkg>/buildtrees/libmodplug/install-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/libmodplug/portfile.cmake:22 (vcpkg_cmake_install)
  scripts/ports.cmake:172 (include)



```
<details><summary>.../vcpkg/buildtrees/libmodplug/install-wasm32-emscripten-dbg-out.log</summary>

```
Change Dir: '<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg'

Run Build Command(s): /opt/homebrew/bin/ninja -v -v -j11 install
[1/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_j2b.cpp.o -MF CMakeFiles/modplug.dir/src/load_j2b.cpp.o.d -o CMakeFiles/modplug.dir/src/load_j2b.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_j2b.cpp
[2/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_dsm.cpp.o -MF CMakeFiles/modplug.dir/src/load_dsm.cpp.o.d -o CMakeFiles/modplug.dir/src/load_dsm.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_dsm.cpp
[3/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_669.cpp.o -MF CMakeFiles/modplug.dir/src/load_669.cpp.o.d -o CMakeFiles/modplug.dir/src/load_669.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_669.cpp
[4/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_far.cpp.o -MF CMakeFiles/modplug.dir/src/load_far.cpp.o.d -o CMakeFiles/modplug.dir/src/load_far.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_far.cpp
[5/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_dbm.cpp.o -MF CMakeFiles/modplug.dir/src/load_dbm.cpp.o.d -o CMakeFiles/modplug.dir/src/load_dbm.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_dbm.cpp
[6/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_amf.cpp.o -MF CMakeFiles/modplug.dir/src/load_amf.cpp.o.d -o CMakeFiles/modplug.dir/src/load_amf.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_amf.cpp
[7/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_dmf.cpp.o -MF CMakeFiles/modplug.dir/src/load_dmf.cpp.o.d -o CMakeFiles/modplug.dir/src/load_dmf.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_dmf.cpp
[8/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_ams.cpp.o -MF CMakeFiles/modplug.dir/src/load_ams.cpp.o.d -o CMakeFiles/modplug.dir/src/load_ams.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_ams.cpp
[9/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_it.cpp.o -MF CMakeFiles/modplug.dir/src/load_it.cpp.o.d -o CMakeFiles/modplug.dir/src/load_it.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_it.cpp
[10/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/fastmix.cpp.o -MF CMakeFiles/modplug.dir/src/fastmix.cpp.o.d -o CMakeFiles/modplug.dir/src/fastmix.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp
FAILED: CMakeFiles/modplug.dir/src/fastmix.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/fastmix.cpp.o -MF CMakeFiles/modplug.dir/src/fastmix.cpp.o.d -o CMakeFiles/modplug.dir/src/fastmix.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:678:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  678 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:684:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  684 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:690:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  690 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:696:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  696 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:702:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  702 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:708:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  708 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:714:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  714 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:720:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  720 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:728:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  728 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:734:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  734 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:740:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  740 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:746:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  746 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:752:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  752 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:758:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  758 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:764:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  764 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:770:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  770 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:780:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  780 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:786:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  786 |         SNDMIX_BEGINSAMPLELOOP16
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:299:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP16'
  299 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:792:2: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  792 |         SNDMIX_BEGINSAMPLELOOP8
      |         ^
<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/fastmix.cpp:291:2: note: expanded from macro 'SNDMIX_BEGINSAMPLELOOP8'
  291 |         register MODCHANNEL * const pChn = pChannel;\
      |         ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[11/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_mtm.cpp.o -MF CMakeFiles/modplug.dir/src/load_mtm.cpp.o.d -o CMakeFiles/modplug.dir/src/load_mtm.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_mtm.cpp
[12/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_mdl.cpp.o -MF CMakeFiles/modplug.dir/src/load_mdl.cpp.o.d -o CMakeFiles/modplug.dir/src/load_mdl.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_mdl.cpp
[13/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_okt.cpp.o -MF CMakeFiles/modplug.dir/src/load_okt.cpp.o.d -o CMakeFiles/modplug.dir/src/load_okt.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_okt.cpp
[14/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_mod.cpp.o -MF CMakeFiles/modplug.dir/src/load_mod.cpp.o.d -o CMakeFiles/modplug.dir/src/load_mod.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_mod.cpp
[15/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_med.cpp.o -MF CMakeFiles/modplug.dir/src/load_med.cpp.o.d -o CMakeFiles/modplug.dir/src/load_med.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_med.cpp
[16/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_mt2.cpp.o -MF CMakeFiles/modplug.dir/src/load_mt2.cpp.o.d -o CMakeFiles/modplug.dir/src/load_mt2.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_mt2.cpp
[17/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_psm.cpp.o -MF CMakeFiles/modplug.dir/src/load_psm.cpp.o.d -o CMakeFiles/modplug.dir/src/load_psm.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_psm.cpp
[18/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_pat.cpp.o -MF CMakeFiles/modplug.dir/src/load_pat.cpp.o.d -o CMakeFiles/modplug.dir/src/load_pat.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_pat.cpp
[19/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_mid.cpp.o -MF CMakeFiles/modplug.dir/src/load_mid.cpp.o.d -o CMakeFiles/modplug.dir/src/load_mid.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_mid.cpp
[20/36] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DHAVE_SETENV -DHAVE_SINF -DHAVE_STDINT_H -DMODPLUG_BUILD -DMODPLUG_STATIC -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src -I<path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/libmodplug -I<path to vcpkg>/buildtrees/libmodplug/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/modplug.dir/src/load_abc.cpp.o -MF CMakeFiles/modplug.dir/src/load_abc.cpp.o.d -o CMakeFiles/modplug.dir/src/load_abc.cpp.o -c <path to vcpkg>/buildtrees/libmodplug/src/70165da81d-0459f39d01.clean/src/load_abc.cpp
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/libmodplug/portfile.cmake
+++ b/ports/libmodplug/portfile.cmake
@@ -17,7 +17,16 @@ vcpkg_from_github(
         005-fix-install-paths.patch # https://github.com/Konstanty/libmodplug/pull/61
 )
 
-vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")
+set(EXTRA_OPTIONS)
+
+if(VCPKG_TARGET_IS_EMSCRIPTEN)
+    list(APPEND EXTRA_OPTIONS "-DCMAKE_CXX_STANDARD=11")
+endif()
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS ${EXTRA_OPTIONS}
+)
 
 vcpkg_cmake_install()
 

--- a/ports/libmodplug/vcpkg.json
+++ b/ports/libmodplug/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libmodplug",
   "version": "0.8.9.0",
-  "port-version": 10,
+  "port-version": 11,
   "description": "The ModPlug mod file playing library.",
   "homepage": "https://github.com/Konstanty/libmodplug",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4606,7 +4606,7 @@
     },
     "libmodplug": {
       "baseline": "0.8.9.0",
-      "port-version": 10
+      "port-version": 11
     },
     "libmorton": {
       "baseline": "0.2.12",

--- a/versions/l-/libmodplug.json
+++ b/versions/l-/libmodplug.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d9af89d39aa5f14bf7314ebc51bc74df26ff2a3b",
+      "version": "0.8.9.0",
+      "port-version": 11
+    },
+    {
       "git-tree": "605d260810f9d1a489af7fb484d509581b39763d",
       "version": "0.8.9.0",
       "port-version": 10

